### DISCUSSION TITLE
Ways of learning

### DISCUSSION THREAD

**Discussion by Dafrill** [codewars/codewars.com#2966] [General]
Hi guys,
I was wandering if you learn programming only by doing Codewars excercises or maybe you use more sources?
(Especially if it comes to Java and C++)

**Comment by Yeoma7353** [comment]
Hello, 
I am also newbie in programming. I use different sources besides Codewars to learn. If I am sitting in bus or train I use Sololearn. They also have a coach code. It is similar to codewars, you also need to solve a task.

My main source is Coursera. They have courses from Google, Meta and universities.

And lastly I read books about it in my mother language.

**Comment by ElDaldoul** [comment]
Books are very usefull, youtube videos of uni professors also teach well.
